### DISCUSSION TITLE
How to debug the wal-redo process？

### DISCUSSION THREAD

**Discussion by softhuafei** [neondatabase/neon#6198] [Q&A]
Hello, thank you for your work. I am a database enthusiast and would like to learn about the implementation details of Neon. However, when I debug the wal-redo process using lldb on macOS, it quickly receives a SIGKILL signal and exits. Could you please advise me on how to debug the wal-redo process? Or do you have any other suggestions for developers?

**Comment by knizhnik** [comment]
> Hello, thank you for your work. I am a database enthusiast and would like to learn about the implementation details of Neon. However, when I debug the wal-redo process using lldb on macOS, it quickly receives a SIGKILL signal and exits. Could you please advise me on how to debug the wal-redo process? Or do you have any other suggestions for developers?

I can not propose better solution rather than temporary increasing `wal_redo_timeout` to some huge value enough for debugging

**Reply by softhuafei** [reply]
Thank you for the suggestion, but I'm still receiving SIGKILL. I tried setting `wal_redo_timeout` to 3 hours, but when I lldb the wal-redo process, it exits after only a few steps. Is there another mechanism that could be killing the wal-redo process?

**Comment by knizhnik** [comment]
There is also `maybe_quiesce` method which tries to shutdown walredo processes which have not perform any redo during `compaction_period*10`
